crates/configuration/src/schema/mod.rs

```diff
@@ -42,4 +42,23 @@ impl Schema {
             .iter()
             .map(|(name, field)| WithNameRef::named(name, field))
     }
+
+    /// Unify two schemas. Assumes that the schemas describe mutually exclusive sets of collections.
+    pub fn merge(schema_a: Schema, schema_b: Schema) -> Schema {
```

Comment:
Would it be helpful to emit errors if there are conflicting collection or object type names? I suppose if they're auto-generated it's not likely to come up